### PR TITLE
ENH: Added r_map option to RPhiBinnedStatistic

### PR DESCRIPTION
**Pull request by jrmlhermitte**
Same idea as #508 .
This is not ready yet. Will be a good opportunity to finally add tests to `RPhiBinnedStatistic`...

Only `r_map` can be overridden but not `phi_map`. We could add it, but I anticipate this to be only useful for the cases of Ewald curvature on 2D data sets.

### CODE REVIEW

File: skbeam/core/accumulators/tests/test_binned_statistic.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from numpy.testing import assert_array_equal, assert_array_almost_equal
 import numpy as np
 import scipy.stats
-from ...utils import bin_edges_to_centers
+from skbeam.core.utils import bin_edges_to_centers
```
Comment (ordirules) [inline skbeam/core/accumulators/tests/test_binned_statistic.py]: I personally don't think test files should do relative imports. Testing should occur from outside.

**Comment by ordirules** [comment]
I kind of would like to have this in (if accepted) before adding the much needed tests for `RPhiBinnedStatistic` which have not been added. Can someone review? @licode?

thanks!

File: skbeam/core/accumulators/tests/test_binned_statistic.py (hunk 1 above)
Comment (ordirules) [reply skbeam/core/accumulators/tests/test_binned_statistic.py]: would like an opinion from you @danielballan . @CJ-Wright  also suggested this should be a relative import, i agree more with him now as a test should be guaranteed to test the correct instance

**Comment by licode** [comment]
Sure, I will take a look.

**Comment by ordirules** [comment]
Thanks. I also posted adding more tests as an issue (#521). It's a minute detail but I think worth the effort.

File: skbeam/core/accumulators/binned_statistic.py [hunk 2]
```diff
@@ -572,6 +572,9 @@ def __init__(self, shape, bins=10, range=None,
         mask : 2-dimensional np.ndarray of ints, optional
             array of zero/non-zero values, with shape `shape`.
             zero values will be ignored.
+        r_map : the map of pixel radii for each pixel. This is useful when the
```
Comment (licode) [inline skbeam/core/accumulators/binned_statistic.py]: add line to explain data type of r_map, like 2-dimensional np.ndarray, optional?

Comment (licode) [reply skbeam/core/accumulators/binned_statistic.py]: r_map can be used to define the radius of each pixel relative to the origin in reciprocal space, like Ewald sphere?

File: skbeam/core/accumulators/tests/test_binned_statistic.py (hunk 1 above)
Comment (licode) [reply skbeam/core/accumulators/tests/test_binned_statistic.py]: I think either one is ok.

File: skbeam/core/accumulators/binned_statistic.py (hunk 2 above)
Comment (jrmlhermitte) [reply skbeam/core/accumulators/binned_statistic.py]: thanks. and yes, I like your wording better. I have added it in if that's okay. thanks!

File: skbeam/core/accumulators/tests/test_binned_statistic.py (hunk 1 above)
Comment (jrmlhermitte) [reply skbeam/core/accumulators/tests/test_binned_statistic.py]: I decided to leave the global import finally. The tests are run in a controlled environment anyway so it shouldn't affect it. The bonus of global imports is you can run this as a script in ipython. Sometimes a little quicker for debugging, at least for me.
(I know there's the pytest --pdb option but sometimes I find it convenient to modify the script and have it return variables and play with them further, plot things etc.)

**Comment by licode** [comment]
LGTM.

**Comment by jrmlhermitte** [comment]
thanks @licode !

**Pull request merged**